include/circt/Dialect/Calyx/CalyxPrimitives.td

```diff
@@ -18,6 +20,42 @@ class CalyxPrimitive<string mnemonic, list<Trait> traits = []> :
     let skipDefaultBuilders = 1;
 }
 
+def ConstantOp: CalyxOp<"constant",
+    [Pure, ConstantLike, FirstAttrDerivedResultType,
+    DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>,
+    AllTypesMatch<["value", "result"]>
+    ]> {
+  let summary = "integer or floating point constant";
+  let description = [{
+    The `constant` operation produces an SSA value equal to some integer or
+    floating-point constant specified by an attribute.
+
+    Example:
+
+    ```
+    // Integer constant
+    %1 = calyx.constant 42 : i32
+
+    // Floating point constant
+    %1 = calyx.constant 42.00+e00 : f32
```

Comment:
In Calyx, we usually treat constants as bit vectors. I'm a bit hesitant to introduce a typed constant here, e.g., how would you convert this when emitting to native Calyx?

Reply:
Aside from "bitnum" and "fixed_point", we can also add a "floating_point" type, which will be interpreted as regular, decimal, floating point values.
But I agree that in the end of the day, everything is just 0 and 1 bit vectors.

Reply:
Those are just interpretations of the bit vectors. We have a native Calyx emitter in CIRCT; how would you convert, e.g., `%1 = calyx.constant 1.1 : f32` using the emitter?

In any case, we should update the emitter and a [test](https://github.com/llvm/circt/blob/main/test/Dialect/Calyx/emit.mlir) for these Calyx constants.

Reply:
You are right, just added the emitter!